docs(components): add vehicle dimensions page

xmfcx opened the pull request:
## Description

<!-- Write a brief description of this PR. -->

Created visualizations for making it easy for people to understand how we name the dimensions of the vehicles. And how the coordinates are placed.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The Reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

kenji-miyake commented on docs/design/autoware-interfaces/components/vehicle-dimensions.md:
```diff
@@ -0,0 +1,68 @@
+
+The distance between front and rear axles.
+
+### track_width
```
Will you add this and some other variables to `vehicle_info`?
https://github.com/autowarefoundation/autoware.universe/blob/c555767cea9b1d75217597f6cd3a0b2faefe9439/vehicle/vehicle_info_util/include/vehicle_info_util/vehicle_info.hpp

yukkysaito replied:
Some comments :pray: 
- Currently it is wheel_tread, not track_width.
- I don't know with whether to include mirrors in left, right_overhang. Currently it is ambiguous.
- I would also like a description of the tires. Mainly used for dead reckoning.

xmfcx replied:
@kenji-miyake 

So, I've ported `vehicle_info` to Autoware.Auto before: https://gitlab.com/autowarefoundation/autoware.auto/AutowareAuto/-/merge_requests/1150/diffs#4054286b653c00d4ad6fddafde60ef1acc5a14a0 and I'm familiar with it.

I've looked for how anyone else names the dist between left and right wheel centers:
- https://en.wikipedia.org/wiki/Axle_track
- https://itstillruns.com/measure-track-vehicle-5850001.html
- https://www.sodialed.com/rc-setup-tips/track-width
And found no results for wheel tread. So I think we should rename `wheel_tread` to `track_width` in the vehicle_info and anything that uses it.

@yukkysaito said:
> I don't know with whether to include mirrors in left, right_overhang. Currently it is ambiguous.

I'm guessing overhangs are only for collider box/prism of the vehicle for obstacle avoidance.

If so, it is better to include it for the minimum bounding box/prism.

If not, could you explain why it exists?

@yukkysaito said:
> I would also like a description of the tires. Mainly used for dead reckoning.

I can add a small illustration for it too, thanks.

yukkysaito replied:
@xmfcx 
>I'm guessing overhangs are only for collider box/prism of the vehicle for obstacle avoidance.
If so, it is better to include windows and any other protrusions for the minimum bounding box/prism.
If not, could you explain why it exists?

The left and right overhangs are used to determine collision and to filter the point clouds hitting the ego vehicle.
There are two issues when mirror is included.
- The size of the vehicle will be larger than the actual vehicle size, making it impossible to drive on narrow roads. This has a significant impact on long vehicles such as buses.
- It would not be able to detect people near the vehicle. (nitpicky)

There are two issues when mirror is not included.
- might hit the mirror of the car.
- The point cloud hits the mirror and there are obstacles near the ego vehicle.


The current implementation does not include mirror, but has a special mirror yaml to filter the point cloud.
https://github.com/autowarefoundation/sample_vehicle_launch/blob/main/sample_vehicle_description/config/mirror.param.yaml
However, there is a possibility of hitting the mirror.

yukkysaito replied:
![image (1)](https://user-images.githubusercontent.com/8327598/162957931-faa9e48b-133f-40e1-8857-1113a00150bb.png)
There are also mirrors like this that are asymmetrical and very long on one side.
This make us worry :cry:

xmfcx replied:
@xmfcx update with the polygon footprint around the vehicle.

kenji-miyake reviewed (commented):
Wow, super cool!
![image](https://user-images.githubusercontent.com/31987104/162857035-c52f3662-b072-4e56-bc4c-0bcf666e39c4.png)

Maybe @yukkysaito has some comments? I'll add him to the reviewers.

xmfcx commented:
@yukkysaito @kenji-miyake could you review again?
- I've added polygon footprint to solve the issues with oversized bounding boxes. 
- Added wheel size information.
Any thoughts?

![screencapture-autowarefoundation-github-io-autoware-documentation-pr-79-design-autoware-interfaces-components-vehicle-dimensions-2022-04-21-18_27_47](https://user-images.githubusercontent.com/10751153/164493591-d9d694a9-6a6e-43b2-9c7c-4218b9922984.png)

yukkysaito commented:
@xmfcx Thank you very much. It's great.
Currently polygonfootprint is not used and defined in autoware, but is it going to be defined it?

LalithVipulananthan commented on docs/design/autoware-interfaces/components/vehicle-dimensions.md:
```diff
@@ -0,0 +1,88 @@
+
+![Vehicle Axes](images/vehicle_axes.svg){: style="width:500px"}
+
+`base_link` frame is in the center of the rear axle.
+`base_link` used very frequently throughout the stack.
```
Minor rewording suggestion:

```suggestion
The `base_link` frame is in the center of the rear axle - this frame is used very frequently throughout the stack.
```

LalithVipulananthan commented on docs/design/autoware-interfaces/components/vehicle-dimensions.md:
```diff
@@ -0,0 +1,88 @@
+
+Overhangs are part of the minimum safety box calculation.
+
+While measuring overhangs, take side mirrors, protruding sensors, wheels to the consideration.
```
```suggestion
When measuring overhangs, side mirrors, protruding sensors and wheels are taken into consideration.
```

LalithVipulananthan commented on docs/design/autoware-interfaces/components/vehicle-dimensions.md:
```diff
@@ -0,0 +1,88 @@
+
+#### left_overhang
+
+The distance between axis centers of left wheels and the left-most point of the vehicle.
```
```suggestion
The distance between the axis centers of the left wheels and the left-most point of the vehicle.
```

LalithVipulananthan commented on docs/design/autoware-interfaces/components/vehicle-dimensions.md:
```diff
@@ -0,0 +1,88 @@
+
+#### right_overhang
+
+The distance between axis centers of right wheels and the right-most point of the vehicle.
```
```suggestion
The distance between the axis centers of the right wheels and the right-most point of the vehicle.
```

LalithVipulananthan commented on docs/design/autoware-interfaces/components/vehicle-dimensions.md:
```diff
@@ -0,0 +1,88 @@
+
+#### front_overhang
+
+The distance between front axle and the foremost point of the vehicle.
```
```suggestion
The distance between the front axle and the foremost point of the vehicle.
```

LalithVipulananthan commented on docs/design/autoware-interfaces/components/vehicle-dimensions.md:
```diff
@@ -0,0 +1,88 @@
+
+#### rear_overhang
+
+The distance between rear axle and the rear-most point of the vehicle.
```
```suggestion
The distance between the rear axle and the rear-most point of the vehicle.
```

LalithVipulananthan commented on docs/design/autoware-interfaces/components/vehicle-dimensions.md:
```diff
@@ -0,0 +1,88 @@
+
+The distance between rear axle and the rear-most point of the vehicle.
+
+### vehicle_length
+
+Total length of the vehicle.
+
+Also calculated by `front_overhang + wheelbase + rear_overhang`
+
+### vehicle_width
+
+Total width of the vehicle.
+
+Also calculated by `left_overhang + track_width + right_overhang`
```
Are the vehicle length and vehicle width something that have their own configuration values, or are they calculated as described above? If they are calculated values, then these lines should be reworded slightly.

xmfcx replied:
```suggestion
### vehicle_length

Total length of the vehicle.

It is calculated by `front_overhang + wheelbase + rear_overhang`

### vehicle_width

Total width of the vehicle.

It is calculated by `left_overhang + track_width + right_overhang`
```

These are calculated automatically from those. Would this work?

LalithVipulananthan replied:
@xmfcx Apologies for the delayed response. Perhaps something like this?

```suggestion
### vehicle_length

Total length of the vehicle. Calculated by `front_overhang + wheelbase + rear_overhang`

### vehicle_width

Total width of the vehicle. Calculated by `left_overhang + track_width + right_overhang`
```

LalithVipulananthan commented on docs/design/autoware-interfaces/components/vehicle-dimensions.md:
```diff
@@ -0,0 +1,88 @@
+
+#### wheel_width
+
+Lateral width of a wheel tire, primarily used for dead reckoning.
```
```suggestion
The lateral width of a wheel tire, primarily used for dead reckoning.
```

LalithVipulananthan commented on docs/design/autoware-interfaces/components/vehicle-dimensions.md:
```diff
@@ -0,0 +1,88 @@
+
+#### wheel_radius
+
+Radius of the wheel, primarily used for dead reckoning.
```
```suggestion
The radius of the wheel, primarily used for dead reckoning.
```

LalithVipulananthan commented on docs/design/autoware-interfaces/components/vehicle-dimensions.md:
```diff
@@ -0,0 +1,88 @@
+
+The polygon defines the minimum collision area for the vehicle.
+
+The points should be ordered clockwise, with origin on the `base_link`.
```
```suggestion
The points should be ordered clockwise, with the origin on the `base_link`.
```

LalithVipulananthan commented on docs/design/autoware-interfaces/components/vehicle-dimensions.md:
```diff
@@ -0,0 +1,88 @@
+
+## Wheel orientations
+
+If the vehicle is going forward, a positive wheel angle will result in vehicle turning left.
```
```suggestion
If the vehicle is going forward, a positive wheel angle will result in the vehicle turning left.
```

LalithVipulananthan commented on docs/design/autoware-interfaces/components/vehicle-dimensions.md:
```diff
@@ -0,0 +1,88 @@
+
+## Notice
+
+Vehicle used in the illustrations created by xvlblo22 and is from <https://www.turbosquid.com/3d-models/modular-sedan-3d-model-1590886>
```
```suggestion
The vehicle used in the illustrations was created by xvlblo22 and is from <https://www.turbosquid.com/3d-models/modular-sedan-3d-model-1590886>.
```

LalithVipulananthan reviewed (commented):
Added some minor grammatical fixes.

xmfcx commented:
@yukkysaito @LalithVipulananthan @kenji-miyake @TakaHoribe could you review it please? I think everything is fine now.

yukkysaito reviewed (approved):
LGTM
